posix/posix.c

```diff
@@ -2437,6 +2437,20 @@ void posix_died(pid_t pid, int exit)
 		proc_threadBroadcast(&ppinfo->wait);
 		proc_lockClear(&ppinfo->lock);
 
+		init = pinfo_find(1);
+		if (init != NULL) {
+			proc_lockSet2(&pinfo->lock, &init->lock);
+			while (pinfo->children != NULL) {
+				cinfo = pinfo->children;
+				cinfo->parent = 1;
```

Comment:
I think you should also take `cinfo->lock` here to safely change `cinfo->parent`. It may race with `posix_getppid()` (which also needs to be fixed and take the lock).

Reply:
Ack, I'd leave a short comment in the code with the explanation.

Reply:
Done